src/main/java/atmosfears/AtmosFearsBE/controller/TimeSeriesController.java

```diff
@@ -0,0 +1,108 @@
+package atmosfears.AtmosFearsBE.controller;
+
+import atmosfears.AtmosFearsBE.database.Particulate;
+import atmosfears.AtmosFearsBE.database.SensorCode;
+import atmosfears.AtmosFearsBE.model.AggregatedParticulates;
+import atmosfears.AtmosFearsBE.model.AirParticulates;
+import atmosfears.AtmosFearsBE.service.AirParticulatesService;
+import lombok.AccessLevel;
+import lombok.experimental.FieldDefaults;
+import org.json.JSONObject;
+import org.springframework.format.annotation.DateTimeFormat;
+import org.springframework.web.bind.annotation.*;
+
+import java.text.SimpleDateFormat;
+import java.util.*;
+
+@RestController
+@FieldDefaults(level = AccessLevel.PRIVATE)
+@RequestMapping("/api/timeseries")
+public class TimeSeriesController {
+
+    final AirParticulatesService airParticulatesService;
+
+    public TimeSeriesController(AirParticulatesService airParticulatesService) {
+        this.airParticulatesService = airParticulatesService;
+    }
+
+    @CrossOrigin()
+    @GetMapping(value = "/locations", produces = "application/json")
+    public String stations() {
+        JSONObject jsonObject = new JSONObject();
+        for (SensorCode sensorCode : SensorCode.values()) {
+            JSONObject locationInfo = new JSONObject();
+            locationInfo.put("code", sensorCode.toString());
+            locationInfo.put("name", sensorCode.getAddress());
+            locationInfo.put("latitude", sensorCode.getLatitude());
+            locationInfo.put("longitude", sensorCode.getLongitude());
+            jsonObject.append("locations", locationInfo);
+        }
+        return jsonObject.toString();
+    }
+
+    @CrossOrigin()
+    @GetMapping(value = "/data", produces = "application/json")
+    public String data(
```

Comment:
This whole method is very long. Can't it be divided into a few smaller chunks that will be aggregated in method? It will be easier to read than having only one method responsible for everything.

I think you should also consider changing endpoint name to something more descriptive than `/data`. It returns pollution data with some information to draw a plot. Maybe it will be better to return plot data separately or to change the endpoint name to make it more obvious what it really does.